File: src/DailyTemperatures.java
import java.util.ArrayDeque;
import java.util.Arrays;
import java.util.Deque;


class Main {
    int[] dailyTemperatures(int[] temps) {
        int n = temps.length;
        int[] ans = new int[n];
        Deque<Integer> stack = new ArrayDeque<>();
        for (int i = n - 1; i >= 0; i--) {
            while (!stack.isEmpty() && temps[i] >= temps[stack.peek()]) {
                stack.pop();
            }
            ans[i] = stack.isEmpty() ? 0 : stack.peek() - i;
            stack.push(i);
        }
        return ans;
    }
    
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] temps = {73, 74, 75, 71, 69, 72, 76, 73};
        System.out.println(Arrays.toString(solution.dailyTemperatures(temps)));
    }
}

File: src/DailyTemperatures.py
from typing import List


class Solution:
    def dailyTemperatures(self, temps: List[int]) -> List[int]:
        res = [0] * len(temps)
        stack = []
        for i, t in enumerate(temps):
            while stack and temps[i] > stack[-1][0]:
                popped_temp, popped_index = stack.pop()
                res[popped_index] = i - popped_index
            stack.append((t, i))
        return res

sol = Solution()
print(sol.dailyTemperatures([73, 74, 75, 71, 69, 72, 76, 73]))